int32 num_channels
int32 sample_rate
int32 sample_width
string endianness
